nilearn/group_sparse_covariance.py

```diff
@@ -545,7 +545,7 @@ def fit(self, subjects, y=None):
             _group_sparse_covariance, memory_level=1)(
                 self.covariances_, n_samples, self.alpha,
                 tol=self.tol, max_iter=self.max_iter,
-                verbose=self.verbose - 1, debug=False)
+                verbose=max(0, self.verbose - 1), debug=False)
```

Comment:
Disclaimer I am not a git/github veteran but this seems to me a very strange way for me to proceed. What if you change something in #366, would you add the commit in this branch too so that the diff you mentioned above (https://github.com/bcipolli/nilearn/compare/bcipolli:allow_silence...fix_warnings) still makes sense?

Maybe somebody with more git/github experience than me can share his opninions.

Also I feel that linking the two PRs is not that great because while the stdout clean-up is a great thing (especially let's be honest for the dataset download mocking which was outputting thousands of lines or so) I am less convinced about trying to get rid of *all* the warnings. Sure the warnings that we can fix in our code should be fixed but it seems too much a burden to go and catch all the warnings across a variety of numpy and scipy distributions.